src/java/org/apache/cassandra/io/sstable/Descriptor.java

```diff
@@ -110,6 +110,11 @@ public Descriptor withFormatType(SSTableFormat.Type newType)
         return new Descriptor(newType.info.getLatestVersion(), directory, ksname, cfname, generation, newType);
     }
 
+    public File tmpFileFor(Component component)
```

Comment:
@mfleming Can you review this commit please. This one contains a set of small changes needed for SAI around adding and removing components from the SSTables.